### PR TITLE
Adding, cleaning and improving samples.

### PR DESCRIPTION
**Pull request by devbotas**
## Description
- Samples moved to samples repo.
- Adjusted readme.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by devbotas** [comment]
Not sure what's up with the build error — looks good on my machine... Azure was unable to restore Windows.Devices.Wifi Nuget?..
image

**Comment by josesimoes** [comment]
Thanks for updating/adding new mqtt samples!
All our code samples live in the samples repo. The ones here where leftovers from the original repo...
Can you please move these over there?

**Comment by devbotas** [comment]
Hmm, so TestAppv5 is now obsolete as well?

**Comment by josesimoes** [comment]
> Hmm, so TestAppv5 is now obsolete as well?

It's not that's obsolete, just misplaced.

**Comment by devbotas** [comment]
@Ellerbach , could you clarify how Testappv5 compare to https://github.com/nanoframework/Samples/blob/main/samples/MQTT/AzureMQTT/AzureMQTT , which is also an Azure sample?

I could then move all the stuff to samples repo, and rename accordingly. Although actual AWS Azure testing may be beyond what I can test right now...

**Comment by Ellerbach** [comment]
There are multiple levels of Azure samples:

* Raw using HTTP for example (like the AWS ones)
* AMQP
* MQTT
* Azure SDK based

**Comment by devbotas** [comment]
I am only interested now in MQTT.
image

**Comment by Ellerbach** [comment]
The Test Mqtt applications in this repository are intended to test connection to various MQTT server for v3.1.1 and v5.0.

**Comment by devbotas** [comment]
I have created a pull request in the Samples repo, thus removing all the confusing duplicated files from here. Updated Readme.md accordingly.

File: README.md [hunk 1]
```diff
@@ -160,54 +147,6 @@ private static void MqttConnectionOpened(object sender, ConnectionOpenedEventArg
 }
 ```
```
Comment (josesimoes) [inline README.md:149]: The usage and example are quite all right here. No need to remove. It's independent from having the samples here or at the other repo.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,163 @@
+# editorconfig.org
```
Comment (Ellerbach) [inline .editorconfig:1]: is this file needed?

File: README.md [hunk 3]
```diff
@@ -32,39 +32,26 @@ The binaries are available as a [NuGet package](https://www.nuget.org/packages/n
...
+
+## MQTT v5 Usage
+
+The usage is globally the same whatever version is used. There are some specificities between v3.1.1 and v5.0. The version 5.0 brings more control and additional properties. For convenience, they are all commented with `v5.0 only` in the properties comments. If you're using a v5.0 property with the v3.1 or v3.1.1 protocol, they'll just be ignored.
 
-For the v5.0, you just need to specify the version before the connection:
+For the v5.0, you just need to specify the version before the connection, like this:
 
 ```csharp
-MqttClient mqtt = new MqttClient("test.mosquitto.org", 8883, true, new X509Certificate(CertMosquitto), null, MqttSslProtocols.TLSv1_2);
```
Comment (Ellerbach) [inline README.md:55]: if you remove this, it won't work. Please don't remove code that will break the way the secure server is working.

File: README.md [hunk 4]
```diff
@@ -32,39 +32,26 @@ The binaries are available as a [NuGet package](https://www.nuget.org/packages/n
...
-mqtt.ProtocolVersion = MqttProtocolVersion.Version_5;
+MqttClient mqtt = new MqttClient("test.mosquitto.org");
+mqtt.ProtocolVersion = MqttProtocolVersion.Version_5; // <-- Add this line if you want to use v5 features.
 var ret = mqtt.Connect("nanoTestDevice", true);
-if (ret != MqttReasonCode.Success)
```
Comment (Ellerbach) [inline README.md:58]: why do you want to remove this part?

File: README.md [hunk 5]
```diff
@@ -32,39 +32,26 @@ The binaries are available as a [NuGet package](https://www.nuget.org/packages/n
...
-{
-    Debug.WriteLine($"ERROR connecting: {ret}");
-    mqtt.Disconnect();
-    return;
-}
 ```
 
-Note: in both example, a specific certificate is needed to connect to the Mosquitto server. You will find it in the [sample](./TestMqtt/TestAppv5.0).
```
Comment (Ellerbach) [inline README.md:66]: This is important, please do not remove. Certificate is one of the main source of issue.

**Review by Ellerbach — COMMENTED** [review]
Quite some open questions. In general, I4m not in favor of removing what's important and then won't work if you copy/paste.

File: .editorconfig (hunk 2 above)
Comment (devbotas) [reply .editorconfig:1]: It silently enforces nF coding guidelines, without having to think about it. It also does not interfere with different guidelines in other projects that use .editorconfig. Also, I do not need to import any settings.

So yeah, I think this file is indeed needed very much.

**Comment by devbotas** [comment]
@josesimoes @Ellerbach The reasons I trimmed that section:
- Many places where we have samples: there's a sample in readme.md, a sample in repo itself, and then some samples in Samples repo. That's confusing.
- The paragraph explains intercompatibility between  v3 and v5 usage, so I trimmed the sample to exactly illustrate the point. Previously it was called "here's a basic example", and let me tell you, it was not basic at all...

File: README.md (hunk 5 above)
Comment (devbotas) [reply README.md:66]: Well it is explained in the TestApp5.0, isn't it? Explaining it in a readme (which by the way, is not really copy-paste compatible), and then referring to an application which is something very different, is not a clear way of telling how to do stuff.

File: README.md (hunk 3 above)
Comment (devbotas) [reply README.md:55]: Which code won't work? And why the code that I left won't work?

File: README.md (hunk 4 above)
Comment (devbotas) [reply README.md:58]: Kinda redundant in explaining of how to set client to v5 mode?

File: README.md (hunk 5 above)
Comment (Ellerbach) [reply README.md:66]: Most people just copy/paste from the README. Please do not remove and change anything on this code. If no certificate, it will fail. Even if the certificate is not present in there, it will trig the fact that it is needed.
So please do not over simplify the docs and sample. You can add more if you want but do not remove. Thanks a lot for your understanding.

**Comment by Ellerbach** [comment]
> * Many places where we have samples: there's a sample in readme.md, a sample in repo itself, and then some samples in Samples repo. That's confusing.

Correct, the original samples long ago were just almost empty README. The sample should be explained. Fill free to add more explanations around the sample by enriching the README. This is a good example of README: https://github.com/nanoframework/nanoFramework.Azure.Devices

You can enrich the samples ones.

> * The paragraph explains intercompatibility between  v3 and v5 usage, so I trimmed the sample to exactly illustrate the point. Previously it was called "here's a basic example", and let me tell you, it was not basic at all...

By over simplifying the code snippet you are creating a problem if the person is trying to make it working. I understand what you want to achieve but there are better ways, for example explaining just the enum without any code around.

**Comment by devbotas** [comment]
I have reverted changes in readme.md. I disagree with the reasons for that, but maybe at least removal of the duplicate samples will be accepted. This would still benefit future lurkers.

@josesimoes, @Ellerbach , please now either accept or reject the PR, and let's move on.

**Comment by josesimoes** [comment]
@Ellerbach please add this to your TODO, so it can be either merged or updated. 😉
